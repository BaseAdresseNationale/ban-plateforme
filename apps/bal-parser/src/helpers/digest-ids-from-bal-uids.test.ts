import { describe, expect, test } from 'vitest';
import {
  idSampleWithAddressId,
  idSampleWithMainTopoId,
  idSampleWithSecondaryTopoId,
  idSampleWithDistrictId,
  idSampleWithAddressIdAndMainTopoId,
  idSampleWithAddressIdAndSecondaryTopoId,
  idSampleWithMainTopoIdAndDistrictId,
  idSampleWithAllIds,
} from './__mocks__/fake-data.js';
import digestIDsFromBalUIDs from './digest-ids-from-bal-uids.js';

describe('digestIDsFromBalUIDs', () => {
  test('should return an empty object for an undefined BanIDs', async () => {
    const id = digestIDsFromBalUIDs();
    expect(id).toMatchSnapshot();
  });
  test('should return an empty object for an empty BanIDs string', async () => {
    const id = digestIDsFromBalUIDs('');
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with only addrID', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithAddressId);
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with only mainTopoID', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithMainTopoId);
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with only secondaryTopoIDs', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithSecondaryTopoId);
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with only districtID', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithDistrictId);
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with addrID and mainTopoID', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithAddressIdAndMainTopoId);
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with addrID and secondaryTopoIDs', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithAddressIdAndSecondaryTopoId);
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with mainTopoID and districtID', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithMainTopoIdAndDistrictId);
    expect(id).toMatchSnapshot();
  });
  test('should return an object with BanID with addrID, mainTopoID, secondaryTopoIDs and districtID', async () => {
    const id = digestIDsFromBalUIDs(idSampleWithAllIds);
    expect(id).toMatchSnapshot();
  });
});
