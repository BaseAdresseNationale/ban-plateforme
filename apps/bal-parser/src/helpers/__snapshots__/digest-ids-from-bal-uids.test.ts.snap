// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`digestIDsFromBalUIDs > should return an empty object for an empty BanIDs string 1`] = `{}`;

exports[`digestIDsFromBalUIDs > should return an empty object for an undefined BanIDs 1`] = `{}`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with addrID and mainTopoID 1`] = `
{
  "addressID": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "mainTopoID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
}
`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with addrID and secondaryTopoIDs 1`] = `
{
  "addressID": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "secondaryTopoIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
}
`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with addrID, mainTopoID, secondaryTopoIDs and districtID 1`] = `
{
  "addressID": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "mainTopoID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "secondaryTopoIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
}
`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with mainTopoID and districtID 1`] = `
{
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "mainTopoID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
}
`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with only addrID 1`] = `
{
  "addressID": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
}
`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with only districtID 1`] = `
{
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
}
`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with only mainTopoID 1`] = `
{
  "mainTopoID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
}
`;

exports[`digestIDsFromBalUIDs > should return an object with BanID with only secondaryTopoIDs 1`] = `
{
  "secondaryTopoIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
}
`;
